src/Xamarin.Android.Build.Tasks/Tests/Xamarin.Android.Build.Tests/Tasks/AndroidComputeResPathsTests.cs

```diff
@@ -99,5 +111,212 @@ public void AndroidComputeResPathsTest (string identity, string logicalName, str
+			var projectDir = Path.Combine (Root, "temp", $"{nameof (XA1031ConflictingFileLinks)}_{hash}");
+			TestOutputDirectories [TestContext.CurrentContext.Test.ID] = projectDir;
+
+			var path = Path.Combine (projectDir, identity1);
+			Directory.CreateDirectory (Path.GetDirectoryName (path));
+			File.WriteAllText (path, contents: "");
+			path = Path.Combine (projectDir, identity2);
+			Directory.CreateDirectory (Path.GetDirectoryName (path));
+			File.WriteAllText (path, contents: "");
+
+			ITaskItem resource1 = new TaskItem (identity1);
+			resource1.SetMetadata ("Link", link1);
+			ITaskItem resource2 = new TaskItem (identity2);
+			resource2.SetMetadata ("Link", link2);
+
+			var intermediateDir = Path.Combine (projectDir, "obj", "Debug", resources ? "res" : "assets");
+			Directory.CreateDirectory (intermediateDir);
+
+			string previousWorking = Environment.CurrentDirectory;
+			try {
+				Environment.CurrentDirectory = projectDir;
+
+				var task = new AndroidComputeResPaths {
+					BuildEngine = engine,
+					ResourceFiles = new ITaskItem [] { new TaskItem (resource1), new TaskItem (resource2) },
+					IntermediateDir = intermediateDir,
+					LowercaseFilenames = resources
+				};
+
+				Assert.IsFalse (task.Execute (), "Task should fail.");
+				Assert.IsTrue (errors.Count > 0, "Task should produce an error.");
+				BuildErrorEventArgs error = errors [0];
+				Assert.AreEqual ("XA1031", error.Code);
+				StringAssert.Contains (resource1.ItemSpec, error.Message);
+				StringAssert.Contains (resource2.ItemSpec, error.Message);
+				StringAssert.Contains (resource1.GetMetadata ("Link"), error.Message);
+				StringAssert.Contains (resource2.GetMetadata ("Link"), error.Message);
+			} finally {
+				Environment.CurrentDirectory = previousWorking;
+			}
+		}
+
+		[Test]
+		public void ManyResourceFiles ()
+		{
+			var projectDir = Path.Combine (Root, "temp", $"{nameof (ManyResourceFiles)}");
+			TestOutputDirectories [TestContext.CurrentContext.Test.ID] = projectDir;
+
+			var items = new List<ITaskItem> (10000);
+			var random = new System.Random (120398);
+			for (var i = 0; i < 10000; i++) {
+				char a = (char) (((int) 'A') + (random.Next (26)) + (32 * i % 2));
+				var identity = $"resource{i}{a}";
+				items.Add (new TaskItem (identity));
+				var path = Path.Combine (projectDir, identity);
+				Directory.CreateDirectory (Path.GetDirectoryName (path));
+				File.WriteAllText (path, contents: "");
+			}
+			var item = items.ToArray ();
+
+			var intermediateDir = Path.Combine (projectDir, "obj", "Debug", "res");
+			Directory.CreateDirectory (intermediateDir);
+
+			var task = new AndroidComputeResPaths {
+				BuildEngine = engine,
+				ResourceFiles = items.ToArray (),
+				IntermediateDir = intermediateDir,
+				LowercaseFilenames = true
+			};
+
+			var stopwatch = System.Diagnostics.Stopwatch.StartNew ();
+			task.Execute ();
+			Assert.Fail (stopwatch.Elapsed.TotalMilliseconds.ToString ());
```

Comment:
This test looks like it would always fail, was this line temporary?

Reply:
Yes indeed. Somehow I pushed the wrong local testing commit to this branch. Let me fix that.